access-control-app/src/cmr/access_control/services/group_service.clj

```diff
@@ -259,11 +259,27 @@
         existing-group (edn/read-string (:metadata existing-concept))]
     (validate-update-group context existing-group updated-group)
     (auth/verify-can-update-group context (assoc existing-group :concept-id concept-id))
-      ;; Avoid clobbering :members by merging the updated-group into existing-group. If updated-group
-      ;; specifies :members then it will overwrite the existing value.
+    ;; Verify the name does not conflict with an existing group
+    (when (not= (:name updated-group) (:name existing-group))
+      (debug (format "Group name update [%s] -> [%s] checking for conflicts"
+                       (:name existing-group)
+                       (:name updated-group)))
+      (when-let [conflicting-concept-id
+                 (->> (search-for-groups context
```

Comment:
This only searches for system groups. What if there are conflict on provider owned groups?